### PR TITLE
[RORDEV-636] Fixes of compilation warnings + unused code removal

### PR DESCRIPTION
**Pull request by mateuszkp96**

### CODE REVIEW

File: integration-tests/src/test/scala/tech/beshu/ror/integration/suites/base/BaseTemplatesSuite.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ trait BaseTemplatesSuite
   private var originComponentTemplateNames: List[String] = List.empty
 
   protected def createIndexWithExampleDoc(documentManager: DocumentManager, index: String): Unit = {
-    adminDocumentManager.createFirstDoc(index, ujson.read("""{"hello":"world"}""")).force()
+    documentManager.createFirstDoc(index, ujson.read("""{"hello":"world"}""")).force()
```
Comment (coutoPL) [inline integration-tests/src/test/scala/tech/beshu/ror/integration/suites/base/BaseTemplatesSuite.scala:47]: the previous one looked better

File: es87x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/BaseDataStreamsEsRequestContext.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ abstract class BaseDataStreamsEsRequestContext[R <: ActionRequest](actionRequest
     backingIndices = {
       val backingIndices = backingIndicesFrom(actionRequest)
       backingIndices match {
-        case BackingIndices.IndicesInvolved(filteredIndices, allAllowedIndices) =>
+        case BackingIndices.IndicesInvolved(filteredIndices, allAllowedIndices@_) =>
```
Comment (coutoPL) [inline es87x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/BaseDataStreamsEsRequestContext.scala:50]: only `_`?

File: es87x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/GetDataStreamEsRequestContext.scala [hunk 3]
```diff
@@ -52,7 +52,7 @@ class GetDataStreamEsRequestContext(actionRequest: GetDataStreamAction.Request,
     ModificationResult.UpdateResponse {
       case r: GetDataStreamAction.Response =>
         blockContext.backingIndices match {
-          case BackingIndices.IndicesInvolved(filteredIndices, allAllowedIndices) =>
+          case BackingIndices.IndicesInvolved(filteredIndices@_, allAllowedIndices) =>
```
Comment (coutoPL) [inline es87x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/GetDataStreamEsRequestContext.scala:55]: the same here

File: es87x/src/main/scala/tech/beshu/ror/es/utils/IndexModuleOps.scala [hunk 4]
```diff
@@ -32,7 +32,7 @@ import scala.language.implicitConversions
 
 class IndexModuleOps(indexModule: IndexModule) {
 
-  def overwrite(readerWrapper: ReaderWrapper): Unit = {
+  def overwrite(): Unit = {
```
Comment (coutoPL) [inline es87x/src/main/scala/tech/beshu/ror/es/utils/IndexModuleOps.scala:35]: let's use the previous version, and do:
```
indexModule.setReaderWrapper(readerWrapper)```

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/scala/tech/beshu/ror/integration/suites/base/BaseTemplatesSuite.scala (hunk 1 above)
Comment (mateuszkp96) [reply integration-tests/src/test/scala/tech/beshu/ror/integration/suites/base/BaseTemplatesSuite.scala:47]: The `documentManager` is passed to a method as a param and the method has more than 50 usages (for all cases `adminDocumentManager` is passed). I can remove `documentManager` from the param list, but IMO it won't improve readability

File: core/src/main/scala/tech/beshu/ror/accesscontrol/factory/RawRorConfigBasedCoreFactory.scala [hunk 5]
```diff
@@ -297,7 +296,7 @@ class RawRorConfigBasedCoreFactory()
         ldapServices <- LdapServicesDecoder.ldapServicesDefinitionsDecoder(ldapConnectionPoolProvider)
         rorKbnDefs <- AsyncDecoderCreator.from(RorKbnDefinitionsDecoder.instance())
         impersonationDefinitionsDecoderCreator = new ImpersonationDefinitionsDecoderCreator(
-          caseMappingEquality, authenticationServices, authProxies, jwtDefs, ldapServices, rorKbnDefs, mocksProvider
+          caseMappingEquality, authenticationServices, authProxies, jwtDefs, ldapServices, mocksProvider
```
[the remark this answers is not in the document]
Comment (mateuszkp96) [reply core/src/main/scala/tech/beshu/ror/accesscontrol/factory/RawRorConfigBasedCoreFactory.scala]: It seems that `jwtDefs` are not needed in the case of the Impersonation feature (not supported now), so I removed it from here and adjusted the code where it's needed

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**